include/transmem/transmem.h

```diff
@@ -153,73 +193,78 @@ friend class GraphMLWriter;
-     * @return matrix encoding the requested transformation
-     * @throws InvalidArgumentException
-     * @throws NoSuchLinkFoundException
-     */
-    QMatrix4x4 getLink(const FrameID &srcFrame, const FrameID &fixFrame, const FrameID &destFrame, const Timestamp &tstamp1, const Timestamp &tstamp2) const;
-
-    /**
-     * @fn QQuaternion getBestLink(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp) const
-     * @brief Returns the best transformation betwen @a srcFrame and @a destFrame.@n@n
-     * Best transformation:@n
+     * @fn StampedAndRatedTransformation getBestLink(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp) const
+     * @brief Returns the best transformation betwen @a srcFrame and @a destFrame encoded in a StampedAndRatedTransformation object.@n@n
+     * Best transformatio:@n
      * Let L1,..,Ln be the links required for the transformation from @a srcFrame to @a destFrame.@n
      * Let Eij be the j-th transformaton entry inserted at time tij in the link Li.@n
      * The best transformation is the transformation obtained at time t, were t minimizes the sum over all |t-tij| from 1 to n.@n@n
      * The resolution of t is 10 ms.
-     * @param srcFrame identifier of the source frame
-     * @param destFrame identifier of the destination frame
-     * @param tstamp returns at what time the best transformation was available
-     * @return matrix encoding the requested transformation
+     * Transmem caches the entries. That is, if no link of a certain transformation was updated, the cached transformation is returned.
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param tstamp Returns at what time the best transformation was available.
+     * @return A StampedAndRatedTransformation object encoding the requested transformation and additional quality information.
      * @throws InvalidArgumentException
      * @throws NoSuchLinkFoundException
      */
-    QMatrix4x4 getBestLink(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp) const;
+    StampedAndRatedTransformation getBestLink(const FrameID &srcFrame, const FrameID &destFrame);
 
-    QMatrix4x4 getBestLinkCached(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp);
+    /**
+     * @fn void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence)
+     * @brief Updates the link from @a srcFrame to @a destFrame with @a confidence. If there was never such a link
+     * registered before, a NoSuchLinkFoundException is thrown.
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param confidence Confidence value.
+     * @throws NoSuchLinkFoundException
+     */
+    void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence);
```

Comment:
For arithmetic types such as double, just pass by copy, rather than by const reference, as it is easier and takes similar or less memory (for instance on x86_64, `int` is 4 bytes, but `const int&` is 8).